newIDE/electron-app/app/ModalWindow.js

```diff
@@ -0,0 +1,67 @@
+const electron = require('electron');
+const BrowserWindow = electron.BrowserWindow; // Module to create native browser window.
+const isDev = require('electron-is').dev();
+const ipcMain = electron.ipcMain;
+
+// Generic function to load external editors in a modal window.
+// Keep a global reference of the window object, if you don't, the window will
+// be closed automatically when the JavaScript object is garbage collected.
+let modalWindow = null;
+
+/**
+ * Open a modal window containing an external HTML5 editor
+ */
+
+const loadModalWindow = ({ onReady, devTools, parentWindow, readyChannelName, indexSubPath, relativeWidth=0.7, relativeHeight=0.9 }) => {
+  if (modalWindow) {
+    modalWindow.show();
+    onReady(modalWindow);
+  }
+
+  const windowOptions = {
+    parent: parentWindow,
+    width: Math.floor(parentWindow.getSize()[0] * relativeWidth),
+    height: Math.floor(parentWindow.getSize()[1] * relativeHeight),
+    backgroundColor: '#000000',
```

Comment:
I was thinking, we could make this configurable and set it to white (#FFFFFF) for Jsfx. This would avoid the "flash" from black to white while the editor is loading and make things look more native :)

Reply:
> Btw one thing that I could do is not show the window before the dom has been loaded - this can help the illusion that it loads faster

This can help / or make the button opening the window appear more sluggish. So it's something to test to see how does it feel in real :)

Reply:
@4ian I added it as an option and enabled it for jsfx. See how you prefer it :)